### PR TITLE
[api] Remove ReceiptByAction() in coreservice.go

### LINKED ISSUE #3208: [api] Remove `ReceiptByAction()` in `coreservice.go`
**What would you like to be added**:

**Why is this needed**:
`ReceiptByAction()` is only used in `GetReceiptByAction()` in `grpcserver.go`

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

`ReceiptByAction()` is only used in `GetReceiptByAction()` in `grpcserver.go`

Fixes #3208 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] TestGrpcServer_GetReceiptByActionIntegrity

**Test Configuration**:

- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/coreservice.go [hunk 1]
```diff
@@ -996,6 +980,11 @@ func (core *coreService) ActionsByAddress(addr address.Address, start uint64, co
 	return res, nil
 }
 
+// GetBlockHashByActionHash returns public method for getBlockHashByActionHash
+func (core *coreService) GetBlockHashByActionHash(h hash.Hash256) (hash.Hash256, error) {
+	return core.getBlockHashByActionHash(h)
```
Comment (Liuhaai) [inline api/coreservice.go:985]: No need creates a new func. L989 can be converted into public func

File: api/grpcserver.go [hunk 2]
```diff
@@ -225,14 +226,20 @@ func (svr *GRPCServer) GetReceiptByAction(ctx context.Context, in *iotexapi.GetR
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	receipt, blkHash, err := svr.coreService.ReceiptByAction(actHash)
+	// receipt, blkHash, err := svr.coreService.ReceiptByAction(actHash)
```
Comment (Liuhaai) [inline api/grpcserver.go:229]: remove

File: api/coreservice.go [hunk 3]
```diff
@@ -149,6 +147,8 @@ type (
 		PendingNonce(address.Address) (uint64, error)
 		// ReceiveBlock broadcasts the block to api subscribers
 		ReceiveBlock(blk *block.Block) error
+		// GetBlockHashByActionHash returns public method for getBlockHashByActionHash
```
Comment (Liuhaai) [inline api/coreservice.go:150]: use the comment on L988 here

**Review by Liuhaai — APPROVED** [review]

File: api/coreservice.go [hunk 4]
```diff
@@ -149,6 +147,8 @@ type (
 		PendingNonce(address.Address) (uint64, error)
 		// ReceiveBlock broadcasts the block to api subscribers
 		ReceiveBlock(blk *block.Block) error
+		// GetBlockHashByActionHash returns block hash by action hash
+		GetBlockHashByActionHash(h hash.Hash256) (hash.Hash256, error)
```
Comment (CoderZhi) [inline api/coreservice.go:151]: BlockHashByActionHash

Comment (CoderZhi) [inline api/coreservice.go:151]: BlockHashByActionHash

File: api/grpcserver.go [hunk 5]
```diff
@@ -225,14 +232,19 @@ func (svr *GRPCServer) GetReceiptByAction(ctx context.Context, in *iotexapi.GetR
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	receipt, blkHash, err := svr.coreService.ReceiptByAction(actHash)
+	receipt, err := svr.coreService.ReceiptByActionHash(actHash)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.NotFound, err.Error())
 	}
+	blkHash, err := svr.coreService.GetBlockHashByActionHash(actHash)
```
Comment (CoderZhi) [inline api/grpcserver.go:239]: this is not efficient. you may consider to use `receipt.BlockHeight` and fetch the block hash by block height.

**Review by CoderZhi — CHANGES_REQUESTED** [review]

**Review by CoderZhi — APPROVED** [review]

**Comment by Liuhaai** [comment]
@LuckyPigeon could you resolve the conflict?

**Pull request merged**